tmva/tmva/src/MethodBDT.cxx

```diff
@@ -2765,50 +2765,93 @@ void TMVA::MethodBDT::MakeClassSpecific( std::ostream& fout, const TString& clas
-         }
-         if (fIsHighBkgCut[ivar]){
-            fout << "   if (inputValues["<<ivar<<"] > "<<fHighBkgCut[ivar] <<")  return -1;  // is background preselection cut" << std::endl;
-         }
-         if (fIsHighSigCut[ivar]){
-            fout << "   if (inputValues["<<ivar<<"] > "<<fHighSigCut[ivar]<<")  return  1;  // is signal preselection cut" << std::endl;
+
+   if(GetAnalysisType() == Types::kMulticlass) {
+      fout << "std::vector<Float_t>& ReadBDTG::GetMulticlassValues__( const std::vector<double>& inputValues ) const" << std::endl;
+      fout << "{" << std::endl;
+      fout << "   std::vector<Float_t> *fMulticlassReturnVal = new std::vector<Float_t>();" << std::endl;
+      fout << std::endl;
+      fout << "   UInt_t nClasses = " << DataInfo().GetNClasses() << ";" << std::endl;
+      fout << "   std::vector<Double_t> temp(nClasses);" << std::endl;
+      fout << "   auto forestSize = fForest.size();" << std::endl;
+      fout << "   // trees 0, nClasses, 2*nClasses, ... belong to class 0" << std::endl;
+      fout << "   // trees 1, nClasses+1, 2*nClasses+1, ... belong to class 1 and so forth" << std::endl;
+      fout << "   UInt_t classOfTree = 0;" << std::endl;
+      fout << "   for (UInt_t itree = 0; itree < forestSize; ++itree) {" << std::endl;
+      fout << "      BDTGNode *current = fForest[itree];" << std::endl;
+      fout << "      while (current->GetNodeType() == 0) { //intermediate node" << std::endl;
+      fout << "         if (current->GoesRight(inputValues)) current=(BDTGNode*)current->GetRight();" << std::endl;
+      fout << "         else current=(BDTGNode*)current->GetLeft();" << std::endl;
+      fout << "      }" << std::endl;
+      fout << "      temp[classOfTree] += current->GetResponse();" << std::endl;
+      fout << "      if (++classOfTree == nClasses) classOfTree = 0; // cheap modulo" << std::endl;
+      fout << "   }" << std::endl;
+      fout << std::endl;
+      fout << "   // we want to calculate sum of exp(temp[j] - temp[i]) for all i,j (i!=j)" << std::endl;
+      fout << "   // first calculate exp(), then replace minus with division." << std::endl;
+      fout << "   std::transform(temp.begin(), temp.end(), temp.begin(), [](Double_t d){return exp(d);});" << std::endl;
+      fout << std::endl;
+      fout << "   for(UInt_t iClass=0; iClass<nClasses; iClass++){" << std::endl;
+      fout << "      Double_t norm = 0.0;" << std::endl;
+      fout << "      for(UInt_t j=0;j<nClasses;j++){" << std::endl;
+      fout << "         if(iClass!=j)" << std::endl;
+      fout << "            norm += temp[j] / temp[iClass];" << std::endl;
+      fout << "      }" << std::endl;
+      fout << "      (*fMulticlassReturnVal).push_back(1.0/(1.0+norm));" << std::endl;
+      fout << "   }" << std::endl;
+      fout << std::endl;
+      fout << "   return *fMulticlassReturnVal;" << std::endl;
+      fout << "}" << std::endl;
+   }
+   else {
```

Comment:
} else {: bracket and else on the same line. Contrary to the impression one might get when reading the code, the closing bracket and the else should be on the same line

Reply:
I found a few more of those, I probably should have scanned through the coding convention doc a bit..